lib/vnet/daemon/client_darwin.m

```diff
@@ -0,0 +1,99 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#import <Foundation/Foundation.h>
+#import <ServiceManagement/ServiceManagement.h>
+
+#include "client_darwin.h"
+
+void BundlePath(struct BundlePathResult *result) {
+    // From the docs:
+    // > This method may return a valid bundle object even for unbundled apps.
+    // > It may also return nil if the bundle object could not be created,
+    // > so always check the return value.
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        result->bundlePath = strdup("");
+        return;
+    }
+    
+    result->bundlePath = VNECopyNSString([main bundlePath]);
+}
+
+NSString * DaemonLabel(void) {
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        return @"";
+    }
+    
+    NSString *bundleIdentifier = [main bundleIdentifier];
+    if ([bundleIdentifier length] == 0) {
+        return bundleIdentifier;
+    }
+    
+    return [NSString stringWithFormat:@"%@.vnetd", bundleIdentifier];
+}
+
+NSString * DaemonPlist(void) {
+    NSString *label = DaemonLabel();
+    if ([label length] == 0) {
+        return label;
+    }
+    
+    return [NSString stringWithFormat:@"%@.plist", label];
+}
+
+void RegisterDaemon(struct RegisterDaemonResult *result) {
+    if (@available(macOS 13, *)) {
```

Comment:
I'm yet to actually verify that those availability checks work on macOS < 13.0, but I have that in my TODO list.

Without them, there's a warning emitted by the compiler which tells you that you ought to wrap calls to `SMAppService` in `@available`. Unfortunately, it seems to be pretty basic and doing an early return doesn't work – the code that's available only on certain versions must be wrapped within an `if (@available(…))`.

I don't think we should be bumping our minimal supported version of macOS just due to a single feature that needs 13.0.